1 Lesson/Homework/Garden.rb

```diff
@@ -0,0 +1,145 @@
+class Raspberry
+  @@states = [:Отсутствует, :Цветение, :Зеленая, :Красная]
+
+  def initialize(index)
+    @index = index
+    @state = @@states[0]
+  end
+
+  def grow!
+    if @state != :Красная
+      @state = @@states[(@@states.find_index(@state) + 1) % @@states.size]
+    end
+  end
+
+  def ripe?
+    @state == :Красная
+  end
+
+  def grab!
+    @state = @@states[0]
+  end
+end
+
+class RaspberryBush
+  attr_reader :berries_quantity
+
+  def initialize(berries_quantity)
+    @berries_quantity = berries_quantity
+    @raspberries = berries_quantity.times.map { |i| Raspberry.new(i) }
+  end
+
+  def grow_all!
+    @raspberries.each { |berry| berry.grow! }
+  end
+
+  def ripe_all?
+    @raspberries.each { |berry| return false unless berry.ripe? }
+    true
+  end
+
+  def give_away_all!
+    @raspberries.each { |berry| berry.grab! }
+  end
+end
+
+class Human
+  attr_reader :name
+  attr_reader :money
+  attr_reader :statistic_of_harvested_berries
```

Comment:
Write it in one line separated by commas

Reply:
Fixed it and decided to implement reading of @statistic_of_harvested_berries in more convenient way
```suggestion
  attr_reader :name, :money
```